### DISCUSSION TITLE
IPAM - create range with all ip included ip addresses

### DISCUSSION THREAD

**Discussion by sadden3194** [netbox-community/netbox#12567] [Help Wanted!] [answered]
Hi all,

maybe I didn't understand the ipam section but I tried this:

IPAM > IP Ranges > Add
![image](https://github.com/netbox-community/netbox/assets/17011498/41443234-2fb5-4b37-8791-2f11712f85c9)

![image](https://github.com/netbox-community/netbox/assets/17011498/99913727-2472-49d5-9cd6-5cc8fd57e838)
But this IP range has no ip addresses and I can't add addreses.

I gues that tha ip range create automatically all addresses vom 192.168.1.1 - 192.168.1.254

Where is my mistake?

**Comment by abhi1693** [comment]
You can add it from IP Address page. It'll automcatically reflect here based on the matching configuration.

On the demo instance, I added this: https://demo.netbox.dev/ipam/ip-addresses/181/

And shows up here: https://demo.netbox.dev/ipam/ip-ranges/5/ip-addresses/

**Comment by sadden3194** [comment]
hmmm okay,
I thought it is enough, if I create a Range and all ip addresses will be create automatically

**Comment by candlerb** [comment]
A range is a separate object type of its own in Netbox, referencing a block of addresses with arbitrary start and and addresses. For example, it could be used to record that a particular range is reserved for a particular purpose ("printers", say). And then you would add individual IP addresses each time you add a printer.

If you want to bulk insert IP addresses, this feature exists under IPAM > IP Addresses > Add. You'll see a tab "bulk create":

![image](https://github.com/netbox-community/netbox/assets/44789/7456889e-f42b-4f71-ba98-782a164bda70)

Click on it. Then you will see:

![image](https://github.com/netbox-community/netbox/assets/44789/8a76dbeb-85e9-4da9-96f5-c3341445d8ce)

**Comment by sadden3194** [comment]
thanks! :)

**Comment by sadden3194** [comment]
is it possible to assign the information, which ip range is which vlan?
or should I write this into the description ?

**Reply by candlerb** [reply, marked as answer]
The association is between Prefix and VLAN.  In your case, you would create a Prefix 192.168.1.0/24 and link it to a VLAN.

It wouldn't make sense to assign individual IPs (or ranges of IPs) each to their own VLAN.
